content/automate/ManageIQ/System/Event/StateMachines/Refresh.class/__methods__/target_refresh.rb

```diff
@@ -0,0 +1,50 @@
+#
+# Description: This method starts a synchronous refresh and save the
+#              task id in the state variables so we can use it when
+#              waiting for the refresh to finish.
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module StateMachines
+          module Refresh
+            class TargetRefresh
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                task_id = event.refresh(refresh_target, true)
+                @handle.set_state_var(:task_id, task_id)
+              end
+
+              private
+
+              def event
+                @handle.root["event_stream"].tap do |event|
+                  if event.nil?
+                    @handle.log(:error, 'Event object is nil')
+                    raise 'Event object not found'
+                  end
+                end
+              end
+
+              def refresh_target
+                target = @handle.object["refresh_target"]
+                @handle.log(:info, "Refresh target: [#{target}]")
+                raise "Refresh target not found" if target.nil?
+                target
```

Comment:
You don't need to do this but since you used `tap` above this feels like another example.

```ruby
@handle.object["refresh_target"].tap do |target|
  @handle.log(:info, "Refresh target: [#{target}]")
  raise "Refresh target not found" if target.nil?
end
```